test_util/test_installer_ccm.py

```diff
@@ -123,153 +83,10 @@ def remote(path):
-        registry_host: string for address where marathon can pull test app
-        test_minuteman: if set to True then test for minuteman service
-        test_dns_search: if set to True, test for deployed mesos DNS app
-        ci_flags: optional additional string to be passed to test
-
-    """
-    marker_args = '-m "not minuteman"'
-    if test_minuteman:
-        marker_args = ''
-
-    run_test_chain = CommandChain('run_test')
-    dns_search = 'true' if test_dns_search else 'false'
-    test_cmd = [
-        'docker', 'run', '-v', '/home/centos/integration_test.py:/integration_test.py',
-        '-e', 'DCOS_DNS_ADDRESS=http://'+dcos_dns,
-        '-e', 'MASTER_HOSTS='+','.join(master_list),
-        '-e', 'PUBLIC_MASTER_HOSTS='+','.join(master_list),
-        '-e', 'SLAVE_HOSTS='+','.join(agent_list),
-        '-e', 'REGISTRY_HOST='+registry_host,
-        '-e', 'DCOS_VARIANT=default',
-        '-e', 'DNS_SEARCH='+dns_search,
-        '-e', 'AWS_ACCESS_KEY_ID='+AWS_ACCESS_KEY_ID,
-        '-e', 'AWS_SECRET_ACCESS_KEY='+AWS_SECRET_ACCESS_KEY,
-        '-e', 'AWS_REGION='+region,
-        '--net=host', 'py.test', 'py.test',
-        '-vv', ci_flags, marker_args, '/integration_test.py']
-    print("To run this test again, ssh to test node and run:\n{}".format(' '.join(test_cmd)))
-    run_test_chain.add_execute(test_cmd)
-
-    check_results(run_loop(ssh_runner, run_test_chain), force_print=True)
-
-
-def prep_hosts(ssh_runner, registry, minuteman_enabled=False):
-    """Runs steps so that nodes can pass preflight checks. Nodes are expected
-    to either use the custom AMI or have install-prereqs run on them. Additionally,
-    Note: break_prereqs is run before this always
-
-    Args:
-        ssh_runner: instance of ssh.ssh_runner.MultiRunner
-        registry: string to configure hosts with trusted registry for app deployment
-        minuteman_enabled: if True, minuteman will be available after DC/OS install
-    """
-    host_prep_chain = CommandChain('host_prep')
-    host_prep_chain.add_execute([
-        'sudo', 'sed', '-i',
-        "'/ExecStart=\/usr\/bin\/docker/ !b; s/$/ --insecure-registry={}:5000/'".format(registry),
-        '/etc/systemd/system/docker.service.d/execstart.conf'])
-    host_prep_chain.add_execute(['sudo', 'systemctl', 'daemon-reload'])
-    host_prep_chain.add_execute(['sudo', 'systemctl', 'restart', 'docker'])
-    host_prep_chain.add_execute(['sudo', 'groupadd', '-g', '65500', 'nogroup'])
-    host_prep_chain.add_execute(['sudo', 'usermod', '-aG', 'docker', 'centos'])
-
-    if minuteman_enabled:
-        host_prep_chain.add_execute(['sudo', 'mkdir', '-p', '/etc/mesosphere/roles'])
-        host_prep_chain.add_execute(['sudo', 'touch', '/etc/mesosphere/roles/minuteman'])
-
-    check_results(run_loop(ssh_runner, host_prep_chain))
+    tunnel.write_to_remote(ip_detect_script, remote('ip-detect.sh'))
+    local_ip = tunnel.remote_cmd(['bash', remote('ip-detect.sh')])
```

Comment:
This method (get_local_address) will not work with clusters which have already been provisioned, since the ip detect script is unique depending on the environment. Can we just default to calling `/opt/mesosphere/bin/detect_ip` and only upload and run a custom ip detect script if the consumer of this method overrides the default behavior by passing in a custom ip detect script? That should make it more generic.